flink/flink1.18-shims/src/main/java/org/apache/zeppelin/flink/Flink118Shims.java

```diff
@@ -0,0 +1,407 @@
+            tableConfig, moduleManager, functionCatalog, catalogManager);
+    Planner planner = (Planner) pair.left;
+    Executor executor = (Executor) pair.right;
+
+    ResourceManager resourceManager = (ResourceManager) createResourceManager(jars, tableConfig);
+
+    return new org.apache.flink.table.api.bridge.scala.internal.StreamTableEnvironmentImpl(catalogManager,
+            moduleManager, resourceManager,
+            functionCatalog, tableConfig, new org.apache.flink.streaming.api.scala.StreamExecutionEnvironment(senv),
+            planner, executor, environmentSettings.isStreamingMode());
+  }
+
+  @Override
+  public Object createJavaBlinkStreamTableEnvironment(Object environmentSettingsObj,
+                                                      Object senvObj,
+                                                      Object tableConfigObj,
+                                                      Object moduleManagerObj,
+                                                      Object functionCatalogObj,
+                                                      Object catalogManagerObj,
+                                                      List<URL> jars,
+                                                      ClassLoader classLoader) {
+    EnvironmentSettings environmentSettings = (EnvironmentSettings) environmentSettingsObj;
+    StreamExecutionEnvironment senv = (StreamExecutionEnvironment) senvObj;
+    TableConfig tableConfig = (TableConfig) tableConfigObj;
+    ModuleManager moduleManager = (ModuleManager) moduleManagerObj;
+    FunctionCatalog functionCatalog = (FunctionCatalog) functionCatalogObj;
+    CatalogManager catalogManager = (CatalogManager) catalogManagerObj;
+    ImmutablePair<Object, Object> pair = createPlannerAndExecutor(
+            classLoader, environmentSettings, senv,
+            tableConfig, moduleManager, functionCatalog, catalogManager);
+    Planner planner = (Planner) pair.left;
+    Executor executor = (Executor) pair.right;
+
+    ResourceManager resourceManager = (ResourceManager) createResourceManager(jars, tableConfig);
+
+    return new StreamTableEnvironmentImpl(catalogManager, moduleManager, resourceManager,
+            functionCatalog, tableConfig, senv, planner, executor, environmentSettings.isStreamingMode());
+  }
+
+  @Override
+  public Object createStreamExecutionEnvironmentFactory(Object streamExecutionEnvironment) {
+    return new StreamExecutionEnvironmentFactory() {
+      @Override
+      public StreamExecutionEnvironment createExecutionEnvironment(Configuration configuration) {
+        return (StreamExecutionEnvironment) streamExecutionEnvironment;
+      }
+    };
+  }
+
+  @Override
+  public Object createCatalogManager(Object config) {
+    return CatalogManager.newBuilder()
+            .classLoader(Thread.currentThread().getContextClassLoader())
+            .config((ReadableConfig) config)
+            .catalogStoreHolder(
+                    CatalogStoreHolder.newBuilder()
```

Comment:
First, thank you for the PR. I was able to compile your branch and it booted up and worked as expected, but the catalog registration did not work in the sense of adding a file catalog store. Looking in the Flink repo, it looks like they started [instantiating the catalog manager differently](https://github.com/apache/flink/blob/release-1.18/flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/bridge/java/internal/StreamTableEnvironmentImpl.java#L132-L152). I don't know if the current flink shim structure is easily adapted to the new utilities added or something else needs to be done. If I get time this weekend I'd like to build off your work to see if we can get it working.